### PR TITLE
extract package comments form and exported comments form

### LINKED ISSUE #105: Separate checks for missing comments and comments format
It's OK when linter tells me I should document a package or a method, or a a constant. But I do not like when it tells me **how** I should document my code.

Would be nice to have separate checks for missing comments and comments format

### PR DESCRIPTION
**Pull request by ademenev**
I do not like when linter tells me how to write my documentation

Closes #105

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -24,9 +24,11 @@ func fail(err string) {
 var defaultRules = []lint.Rule{
 	&rule.VarDeclarationsRule{},
 	&rule.PackageCommentsRule{},
+	&rule.PackageCommentsFormRule{},
```
Comment (mgechev) [inline config.go]: These rules should be also documented in the readme.

File: rule/exported-form.go [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+import (
+	"fmt"
+	"go/ast"
+	"go/token"
+	"strings"
+
+	"github.com/mgechev/revive/lint"
+)
+
+// ExportedFormRule lints given else constructs.
```
Comment (mgechev) [inline rule/exported-form.go]: This comment need to be rewritten

**Comment by ademenev** [comment]
Hmm, one thing I did not think about is breaking existing setups. Possibly a better approach would be to make `exported` and `package-comments` configurable

**Comment by mgechev** [comment]
@ademenev that's a good point, somehow related to a better release process.

If we ignore backward compatibility of the config file, I'd prefer to go with two new rules, as you proposed. The change makes sense because it enforces better modularity and separation of concerns.

My suggestion would be to first address https://github.com/mgechev/revive/issues/100 and tag 0.1.0. After that merge your change (once there are tests for the new rules and we update the docs + the default set of rules to keep it compatible with golint) and tag 1.0.0-rc.0. After that, once 1.0.0-rc.0 has been well tested by developers, we can release 1.0.0.

What are your thoughts?

cc @chavacava

**Comment by ademenev** [comment]
@mgechev I just started using revive, and warnings about my comment style were annoying, so I quickly put together some code to mute them. Now, after thinking more, I gather a better solution would be to not introduce new rules, but instead make existing ones configurable.

So I suggest that you drop this PR, and I will prepare another one, with tests and docs. After 0.1.0 I'ii submit another PR. Would that work?

**Comment by mgechev** [comment]
Why you think making the rules configurable would be a better approach?

**Comment by ademenev** [comment]
The reason is keeping BC.

Introducing separate rules will immediately make files that use disable:export comments generate warning - people expected that to suppress both checks, but only one will be suppressed.

**Comment by mgechev** [comment]
Yes, backwards compatibility is the reason for my proposal: 0.1.0 and after that 1.0.0 after your PR goes in. @chavacava can share his opinion as well.

**Comment by ademenev** [comment]
My point is : making the rules configurable adds desired feature without breaking BC

**Comment by chavacava** [comment]
As pointed by @ademenev, making the rule configurable preserves BC. 
We need to work on #66 to have a clear definition of revive's releases policies

**Comment by mgechev** [comment]
Just to add more context to this comment https://github.com/mgechev/revive/pull/106#issuecomment-460097897

If we introduce SemVer to revive, we can release the current set of changes as 0.1.0. We can release the rule refactoring (i.e., splitting into multiple rules) as 1.0.0. This is fine, because a major release in SemVer allows introducing breaking changes. Everyone who hasn't migrated yet, can rely on 0.1.0 and preserve the rule behavior.

**Pull request closed without merging**